### PR TITLE
 Bschnurr/refactor/test discovery

### PR DESCRIPTION
**Pull request by bschnurr**
Refactoring discovery
-rename postfix files to prefilx pytest/unittest
- Removing stub classes UnitTestDiscoverer and PytestTestDiscoverer that were exported to VS.
- Now Pytest/PytestTestDiscover and UnitTest/UnitTestTestDiscover as the classes that get exported.  They also now inherit from an abstract parent PytonTestDiscoverer
- Removed DiscoveryFactory. No longer since we started exporting separate classes to VS for discovery

### CODE REVIEW

File: Python/Product/TestAdapter.Executor/Common/PythonTestDiscoverer.cs [hunk 1]
```diff
@@ -19,34 +19,10 @@
...
-    /// Note even though we specify  [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)] we still get all .py source files
-    /// from all testcontainers.  
-    /// </summary>
-    [FileExtension(".py")]
-    [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)]
-    public class PytestTestDiscoverer : PythonTestDiscoverer {
-        public PytestTestDiscoverer() : base(TestFrameworkType.Pytest) { 
-        }
-    }
-
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/Common/PythonTestDiscoverer.cs]: No longer export theses.  Now we export the children of this abstract classs (PytestTestDiscoverer and UnitTestTestDiscoverer)

File: Python/Product/TestAdapter.Executor/Common/PythonTestDiscoverer.cs [hunk 2]
```diff
@@ -58,6 +34,13 @@ public abstract class PythonTestDiscoverer : ITestDiscoverer {
             _frameworkType = frameworkType;
         }
 
+        abstract public void DiscoverTests(
+           IEnumerable<string> sources,
+           PythonProjectSettings settings,
+           IMessageLogger logger,
+           ITestCaseDiscoverySink discoverySink
+        );
+
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/Common/PythonTestDiscoverer.cs]: New abstract method that does the actual work.

Note: When running from command line the test executor will also call this, thats why its public

File: Python/Product/TestAdapter.Executor/Pytest/PytestTestDiscoverer.cs [hunk 3]
```diff
@@ -28,17 +28,24 @@
...
+    /// Note even though we specify  [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)] we still get all .py source files
+    /// from all testcontainers.  
+    /// </summary>
+    [FileExtension(".py")]
+    [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)]
+    public class PytestTestDiscoverer : PythonTestDiscoverer {
         private IMessageLogger _logger;
         private static readonly string DiscoveryAdapterPath = PythonToolsInstallPath.GetFile("PythonFiles\\testing_tools\\run_adapter.py");
 
-        public TestDiscovererPytest(PythonProjectSettings settings) {
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/Pytest/PytestTestDiscoverer.cs]: moving settings from a constructor param to a function param. Also simplifies testing by not having to build a settings object

File: Python/Product/TestAdapter.Executor/TestAdapter.Executor.csproj [hunk 4]
```diff
@@ -93,21 +93,19 @@
     <Compile Include="Common\Config\RunSettingsUtil.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="Pytest\PytestExtensions.cs" />
-    <Compile Include="Pytest\TestDiscovererPytest.cs" />
-    <Compile Include="Pytest\TestExecutorPytest.cs" />
-    <Compile Include="Common\Services\DiscovererFactory.cs" />
+    <Compile Include="Pytest\PytestTestDiscoverer.cs" />
+    <Compile Include="Pytest\PytestTestExecutor.cs" />
```
Comment (huguesv) [inline Python/Product/TestAdapter.Executor/TestAdapter.Executor.csproj]: You renamed the file but not the class

File: Python/Product/TestAdapter.Executor/TestAdapter.Executor.csproj [hunk 5]
```diff
@@ -93,21 +93,19 @@
...
+    <Compile Include="Common\Services\CodeCoverage.cs" />   
     <Compile Include="Common\Services\ProcessExecute.cs" />
     <Compile Include="Common\Services\PythonDebugMode.cs" />
     <Compile Include="Common\TestCollection.cs" />
     <Compile Include="Common\PythonTestDiscoverer.cs" />
     <Compile Include="Common\TestProtocol.cs" />
     <Compile Include="Common\TestReader.cs" />
-    <Compile Include="UnitTest\TestExecutorUnitTest.cs" />
-    <Compile Include="UnitTest\TestDiscovererUnitTest.cs" />
+    <Compile Include="UnitTest\UnitTestTestExecutor.cs" />
```
Comment (huguesv) [inline Python/Product/TestAdapter.Executor/TestAdapter.Executor.csproj]: Same here, file was renamed but not the class

Comment (bschnurr) [reply Python/Product/TestAdapter.Executor/TestAdapter.Executor.csproj]: thanks

**Pull request merged**